src/ucp/core/ucp_request.c

```diff
@@ -264,6 +264,84 @@ UCS_PROFILE_FUNC_VOID(ucp_request_memory_dereg,
     }
 }
 
+ucs_status_t ucp_request_rndv_buffer_reg(ucp_request_t *req)
+{
+    ucp_ep_t        *ep = req->send.ep;
+    uct_md_h         md;
+    ucp_lane_index_t lane;
+    ucp_lane_index_t i;
+    ucp_lane_index_t r;
+    ucs_status_t     status;
+
+    req->send.reg_rsc = UCP_NULL_RESOURCE;
+
+    for (i = 0; i < ucp_ep_rndv_num_lanes(ep); i++) {
+        if (ucp_ep_rndv_md_flags(ep, i) & UCT_MD_FLAG_NEED_MEMH) {
+            lane = ucp_ep_get_rndv_get_lane(ep, i);
+            ucs_assert(lane != UCP_NULL_LANE);
```

Comment:
No need for assert, there is already one inside `ucp_ep_get_rndv_get_lane`

Reply:
this is hint to be sure that lane could not be NULL

Reply:
I got that, but still there is the one inside `ucp_ep_get_rndv_get_lane`

Reply:
may be in future this assert will help you to catch critical issue

Reply:
ok, asserts are removed